### ISSUE TITLE
UglyToad.PdfPig.Content.Page.GetWords() is splitting words where it shouldn't.

### ISSUE THREAD

**Issue by chrismaduro** [UglyToad/PdfPig#663]
From the image below you can see by the empty black squares in the pdf that there are no spaces within the words.
![image](https://github.com/UglyToad/PdfPig/assets/10157/6621f855-d856-4971-b998-083e1d73eb40)

However, the library seems to be chopping words into pieces as you can see from the joined word string using a pipe delimiter, below.
![image](https://github.com/UglyToad/PdfPig/assets/10157/83363812-4668-48c3-ba7f-658b04a6d01b)

**Comment by EliotJones** [comment]
Hi @chrismaduro `page.GetWords()` by default uses an extremely naive heuristics based approach to grouping letters into words. This won't work for all documents, especially where the spacing is not typical (i.e. not standard Microsoft Word type text).

It may work to try the more powerful nearest neighbor extractor: https://github.com/UglyToad/PdfPig/wiki/Document-Layout-Analysis#nearest-neighbour-method

If not it is probably necessary to custom write some logic for the files of interest. The default word extractor is a fairly short loop over the letters on the page so can be easily copied and adjusted by consumers: https://github.com/UglyToad/PdfPig/blob/master/src/UglyToad.PdfPig/Util/DefaultWordExtractor.cs

**Comment by chrismaduro** [comment]
Yes, I dug into the documentation to try to figure out what was going on, and changing the algorithm solved the problem. Thank you for mentioning it here. Hopefully in the future this post will help other devs who may have missed that in the docs, like I did. I previously had not realized that a pdf was a collection of glyphs, I thought since pdf editors act much like word processors at time, that the pdf format also stored the information more abstractly (words in a textbox), and that somehow the OCR I performed was chopping the words incorrectly. Reading the docs was very informative.
